src/control/cmd/daos_agent/infocache.go

```diff
@@ -60,17 +60,48 @@ func (aic *attachInfoCache) loadBalance(numaNode int) int {
 	return deviceIndex
 }
 
+// selectRemote is called when no local network adapters
+// was detected on the local NUMA node and we need to pick
+// one attached to a remote NUMA node. Some balancing is
+// still required here to avoid overloading a specific interface
+func (aic *attachInfoCache) selectRemote() (int, int) {
+	aic.mutex.Lock()
```

Comment:
As Tom noted, it would be better to put `defer aic.mutex.Unlock()` here to ensure that the lock is always released even if the code below changes to add multiple exit points.

Reply:
done